lib/passport-amebame/index.d.ts

```diff
@@ -0,0 +1,42 @@
+import passport = require('passport');
+import express = require('express');
+
+interface Profile extends passport.Profile {
+  id: string;
+  displayName: string;
+  imageUrl: string;
+  gender?: string;
+  birthday?: string;
+
+  _raw: string;
+  _json: any;
+}
+
+interface IStrategyOption {
+  clientID: string;
+  clientSecret: string;
+  scope?: string;
+  authOrigin?: string;
+  profileOrigin?: string;
+}
+
+interface IAuthorizationParams {
+  frm_id?: string;
+  ameba_frmId?: string;
+  provider_id?: string;
+  state?: string;
+  redirect_uri?: string;
+  force_provide?: string;
+  response_type?: string;
+  display?: string;
+}
+
+export class Strategy extends passport.Strategy {
+  constructor(options: IStrategyOption,
+      verify: (accessToken: string, refreshToken: string, profile: Profile, done: (error: any, user?: any) => void) => void);
```

Comment:
```suggestion
      verify: (accessToken: string, refreshToken: string, profile: Profile, done: (error: Error | null, user?: any) => void) => void);
```
これでも良さそうですがどうでしょう:eyes:

Reply:
`userProfile`のほうは確実に `Error | null` にできそうかも。

https://github.com/openameba/passport-amebame/pull/8/files#diff-7dabf0d0980f22f62e6cd9ca33a717079d59a1e1305a70021a726dd401a709acR41

Reply:
https://github.com/openameba/passport-amebame/pull/8#discussion_r925339925

ここみるといけそうな気もしなくもないですがどうなんでしょう
verify自体は`OAuth2Strategy`に渡しているだけなのである程度合わせても良さそうな気もしました:eyes:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport-oauth2/index.d.ts#L55

Reply:
[実装的にはtruthy or falsyで判断できれば何でも良さそう](https://github.com/jaredhanson/passport-oauth2/blob/ee3fe9f17c0f3a90f2d9d938f267e9942b9fba49/lib/strategy.js#L184-L185)でライブラリで決められそうなので、なんならuser含めて形付けておきましょか。